server/src/proxy/hotspot.rs

```diff
@@ -0,0 +1,411 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! hotspot recorder
+use std::{fmt::Write, sync::Arc, time::Duration};
+
+use ceresdbproto::storage::{
+    PrometheusQueryRequest, RequestContext, SqlQueryRequest, WriteRequest,
+};
+use common_util::{runtime::Runtime, timed_task::TimedTask};
+use log::{info, warn};
+use serde::{Deserialize, Serialize};
+use spin::Mutex;
+pub use spin::Mutex as SpinMutex;
+use tokio::sync::mpsc::{self, Sender};
+
+use crate::proxy::{hotspot_lru::HotspotLru, util};
+
+type ReadKey = String;
+type WriteKey = String;
+const TAG: &str = "hotspot autodump";
+const RECODER_CHANNEL_CAP: usize = 64 * 1024;
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(default)]
+pub struct Config {
+    /// Max items size for read hotspot
+    read_cap: Option<usize>,
+    /// Max items size for write hotspot
+    write_cap: Option<usize>,
+    dump_interval: Duration,
+    auto_dump: bool,
+    /// Max items for dump hotspot
+    auto_dump_len: usize,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            read_cap: Some(10_000),
+            write_cap: Some(10_000),
+            dump_interval: Duration::from_secs(5),
+            auto_dump: true,
+            auto_dump_len: 10,
+        }
+    }
+}
+
+enum Message {
+    Query(ReadKey),
```

Comment:
Maybe should split to `SqlQuery`, `InfluxqlQuery` and `PromQuery`?
And we just left `SqlQuery` now, because the others are now not supported well?

Reply:
Maybe we can split after we support `InfluxqlQuery` and `PromQuery` well.